### PR TITLE
[MINT-2998] Status route mutation handler

### PR DESCRIPTION
**Pull request by patrickseguraoddball**
# EASI-2998

## Description

- Added mutation handler to parse query param from url/email link and attempt to update status from task list
  - Error states for mutation and if status is already set
  - Success state
  - Clears param after attempt
 - Added default status on `<Status />` component tp take a default status for clearance status from email
- Added success message on next page/task list
- Added cypress test

## How to test this change

- Navigate to task list of a model plan
- Added query param `?model-status=ACTIVE` or any `ModelStatus` enum value.  This is case sensitive.  I decided to leave it case sensitive to stay true to enum value
- Verify status gets updated and alert renders
- Add a random string, verify error
- Add the existing status, verify specific error

## PR Author Checklist

- [x] I have provided a detailed description of the changes in this PR.
- [x] I have provided clear instructions on how to test the changes in this PR.
- [x] I have updated tests or written new tests as appropriate in this PR.
- [x] Updated the [Postman Collection](../MINT.postman_collection.json) if necessary.

## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.

### CODE REVIEW

File: src/views/ModelPlan/TaskList/index.tsx [hunk 1]
```diff
@@ -213,6 +231,77 @@ const TaskList = () => {
...
+              message: t('statusUpdateError', {
+                status: translatedStatus
+              }),
+              status: 'error'
+            });
+            params.delete('model-status');
+            history.replace({
+              search: params.toString()
+            });
+          });
```
Comment (garyjzhao) [inline src/views/ModelPlan/TaskList/index.tsx:277]: getting inaccurate error when i enter `?model-status=CANCEL`

image

**Review by garyjzhao — CHANGES_REQUESTED** [review]
Found one error. see comment

File: src/views/ModelPlan/Status/index.tsx [hunk 2]
```diff
@@ -29,44 +29,59 @@ type StatusFormProps = {
...
+        })
+        .catch(errors => {
+          formikRef?.current?.setErrors(errors);
+        });
+    }
   };
 
   const initialValues: StatusFormProps = {
-    status: status ?? undefined
+    status: modelStatus ?? status ?? undefined
```
Comment (garyjzhao) [inline src/views/ModelPlan/Status/index.tsx:84]: This does not seem right. What are you trying to do here?

Comment (patrickseguraoddball) [reply src/views/ModelPlan/Status/index.tsx:84]: Setting the default to the query param, then the query return, then string

File: src/views/ModelPlan/TaskList/index.tsx (hunk 1 above)
Comment (patrickseguraoddball) [reply src/views/ModelPlan/TaskList/index.tsx:277]: CANCEL isn't a status

Comment (patrickseguraoddball) [reply src/views/ModelPlan/TaskList/index.tsx:277]: We don't have translations for CANCEL and nor would we really ever have that passed

Comment (garyjzhao) [reply src/views/ModelPlan/TaskList/index.tsx:277]: oh right. i was testing a status that does not exist.
Nevertheless, the error banner says `there was an error updating the status to .` <--

Comment (patrickseguraoddball) [reply src/views/ModelPlan/TaskList/index.tsx:277]: Right that will never happen, it also hasn't been mocked.  That text always runs through i18n translation.  I can ask Natasha but I dont thin kthis scenario would ever happen as it's coming from email link.  I'll see if I can get a general error from her.

Comment (patrickseguraoddball) [reply src/views/ModelPlan/TaskList/index.tsx]: @garyjzhao Just ended up adding the raw value if attempting to add something other than a set status

**Pull request merged**